docs/website/contents/for-developers/Whitepaper.md

```diff
@@ -0,0 +1,244 @@
+Improving the selection requires validation of the fetched block (and maybe more, if blocks arrived out of order).
+If the fetched block is invalid, ChainSel disconnects from the upstream peer who sent it, unless that block may have been pipelined; see the specific pipelining rules.
+In turn, if a fetched block should be pipelined, ChainSel signals the ChainSync servers to send that header just before it starts validating that block.
+If it turns out to be invalid, ChainSel promptly signals the ChainSync servers to send the corresponding MsgRollBack.
+
+The combined ledger and protocol state is maintained alongside the local selection by ChainSel, so that blocks can be validated.
+
+## ChainDB
+
+A Praos node must not introduce unnecessary delays between receiving a block and forwarding it along.
+It is therefore an important observation that the ChainDB does not require the Durability property of ACID: upstream peers will always be available to replace any blocks a node loses.
+
+## LedgerDB
+
+In both ChainSel and ChainSync, rollbacks require maintenance of/access to the past kcp+1 states, not only the tip's state — access to any such state must be fast enough to avoid disrupting the semantics of the worst-case delay parameter Delta assumed in the Praos paper's security argument.
+
+In addition to validation in ChainSel and ChainSync, these ledger states are how the node handles a fixed set of queries used by wallets, CLI tools, etc via the LocalStateQuery mini protocol.
+
+## Mempool & TxSubmission
+
+The Mempool maintains a sequence of transactions that could inhabit a hypothetical block that is valid and extends the current local selection.
+The Mempool is bounded via a multi-dimensional notion of size such that it never contains more transactions than could fit in N blocks.
+Peers synchronize their Mempools via the TxSubmission protocol.
+This mini protocol leverages the fact that the Mempool is a sequence as opposed to an unordered set; a simple integer suffices as the iterator state for the TxSub client.
+(Recall that transactions flow from client to server, since the orientation is determined by the flow of blocks and pending transactions naturally flow opposite of blocks.)
+
+## The Mint aka Block Forge
+
+Whenever the wall clock enters a new slot, the node checks whether its configured minting credentials (if any) were elected by the protocol state and forecasted ledger state to lead this slot.
+If so, it mints a block that extends its local selection (or its immediate predecessor if the tip somehow already occupies this slot or greater) and contains the longest prefix of the Mempool's transactions that can fit.
+That block is then sent to ChainSel, as if it had been fetched.
+
+When (if) the node selects that block, the Mempool will react as it does for any change in the local selection: it discards any transactions that are no longer valid in the updated hypothetical block the Mempool continuously represents.
```

Comment:
This is fine, but it alludes to a behavior of the mempool that was not described above, namely resyncing with the ledgerdb.